agent/command/git.go

```diff
@@ -707,13 +712,24 @@ func (c *gitFetchProject) fetchModuleSource(ctx context.Context,
 		repo:     repo,
 		branch:   "",
 		dir:      moduleBase,
+		method:   cloneMethod,
 	}
 	// Module's location takes precedence over the project-level clone
 	// method.
 	if strings.Contains(opts.location, "git@github.com:") {
 		opts.method = evergreen.CloneMethodLegacySSH
+	} else if opts.method == evergreen.CloneMethodAccessToken {
+		if appToken, ok := appTokens[opts.owner]; ok {
```

Comment:
I'm kind of curious how the `else` case ever gets hit, because it seems impossible - [`appTokens` is set in the same case where `opts.method == evergreen.CloneMethodAccessToken`](https://github.com/evergreen-ci/evergreen/blob/2062dece757ef093f0c7948e7b27853368000273/agent/command/git.go#L847-L850), in which case this if block always evaluates to true. Doesn't this devolve into:

```go
else if opts.method == evergreen.CloneMethodAccessToken {
    opts.token = appToken // If we've reached this else-if, this is always going to be set and will never dynamically fetch the installation token?
}
```
}
Am I missing something?

If you're trying to make an in-memory cache of app tokens for modules to save calls to `CreateInstallationToken` for later module clones, you have to move the `appTokens` initialization outside of the module loop.